### ISSUE TITLE
Allow "pushing" and "pulling" as parameters of game:get/set_ability(...)

### ISSUE THREAD

**Issue by Diarandor** [christopho/solarus#788]
Allow to enable/disable the pushing and pulling built-in abilities with the commands game:get/set_ability(...), etc.

This could be necessary when we want to avoid the hero to push/pull blocks or walls in some exceptional cases (while carrying some item/weapon, for instance).

**Comment by MetalES** [comment]
Yeah, a game:set_ability(ability, nil/value) could be an interesting addition, especially for some actions / game type

**Comment by Diarandor** [comment]
I realized that the pulling action can be disabled if we use the function "game:on_command_pressed(command)" to override it (returning true when necessary, to avoid the built-in effect).

Also, I suppose that we can disable the pushing effect in a similar way (although I have not tested this). The idea is to return true in the event when an arrow direction is pressed and there is an obstacle in front of the hero (which can be detected using the function "entity:test_obstacles(dx, dy, [layer])").

So you can close this issue if you want since there are workarounds, although having such functions would be useful to simplify the code.

**Comment by Diarandor** [comment]
I confirm that the pushing effect cannot be disabled with the workaround that uses "game:on_command_pressed(command)", because when you press an arrow direction and hold it pressed, the hero moves but the event is only called when you press the key the first time, so when you reach the wall the event is not being called.

We will need a function game:get/set_ability(...) to disable the pushing effect. (For the pulling effect that would be usefull too.)

**Comment by Diarandor** [comment]
I just want to point out that to make, for instance, a custom sword, it may be necessary to disable temporarily the "pushing", "pulling" (and hence "grabbing") abilities to avoid pushing/pulling/grabbing a block while the animation "sword_tapping" of a custom (non-built-in) sword is played, and there may be similar problems with walls. (I am not having problems with blocks since I am using my custom blocks. But I am still having problems with walls.)
